File: 1217-relative-sort-array/relative-sort-array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        # 1. Count frequencies of elements in arr1
        counts = Counter(arr1)
        res = []
        
        # 2. Add elements in the order of arr2
        for num in arr2:
            if num in counts:
                res.extend([num] * counts[num])
                del counts[num]  # Remove so we know what's left
                
        # 3. Handle remaining elements in sorted order
        remaining = []
        for num, count in counts.items():
            remaining.extend([num] * count)
        remaining.sort()
        
        return res + remaining

File: 1217-relative-sort-array/relative-sort-array.ts
function relativeSortArray(arr1: number[], arr2: number[]): number[] {
    // arrays that are found in arr1 but not in arr2
    let ex_arrs : number[] = [] 
    let count: Record<number, number> = {}
    let result : number[] = []

    for (const num of arr2) {
       count[num] = 0;
    }

    for (const num of arr1) {
        if (count[num] >= 0) {
            count[num] += 1
        }
        if (!arr2.includes(num)) {
            ex_arrs.push(num)
        }
    }

    for (const num of arr2) {
        let cnt = count[num];
        while(cnt >= 1) {
            result.push(num)
            cnt -= 1
        }
    }

    return result.concat(ex_arrs.sort((a, b) => a - b));
};